src/construction/construction.py

```diff
@@ -1,282 +1,81 @@
         try:
-            additions_df = read_csv(additions_filepath)
-            additions_df["instance"] = additions_df["instance"].astype("Int64")
-            constructed_df = apply_additions(main_df, additions_df)
+            construction_df = read_csv(construction_file_path)
         except pd.errors.EmptyDataError:
             construction_logger.warning(
-                f"Additions file {additions_filepath} is empty, skipping..."
+                f"Construction file {construction_file_path} is empty, skipping..."
             )
+            return snapshot_df
+    else:
+        construction_logger.info(
+            "Construction file not found, skipping construction..."
+        )
+        return snapshot_df
 
-    # Save the constructed dataframe as a CSV
-    tdate = datetime.now().strftime("%Y-%m-%d")
-    construction_output_filepath = os.path.join(paths["root"], "construction", f"constructed_snapshot_{tdate}_v{run_id}.csv")
-    write_csv(construction_output_filepath, constructed_df)
-    return constructed_df
-
-
-def apply_amendments(main_df, amendments_df):
-    """Apply amendments to the main snapshot."""
-    key_cols = ["reference", "year", "instance"]
-    numeric_cols = ["219", "220", "242", "243", "244", "245",
-                    "246", "247", "248", "249", "250"]
-    numeric_cols_new = [f"{i}_updated" for i in numeric_cols]
-
-    accepted_amendments_df = amendments_df.drop(amendments_df[~amendments_df.accept_changes].index)
-
-    if accepted_amendments_df.shape[0] == 0:
-        construction_logger.info("Amendments file contained no records marked for inclusion")
-        return main_df
-
-    # Drop the diff columns
-    accepted_amendments_df = accepted_amendments_df.drop(columns=[col for col in accepted_amendments_df.columns if col.endswith("_diff")])
-
-    # Join the amendments onto the main snapshot
-    amended_df = pd.merge(main_df, accepted_amendments_df, how="left", on=key_cols)
-
-    # Drop the old numeric cols and rename the amended cols
-    amended_df = amended_df.drop(columns=numeric_cols)
-    cols_to_rename = dict(zip(numeric_cols_new, numeric_cols))
-    amended_df = amended_df.rename(columns=cols_to_rename, errors="raise")
+    # Make a copy of the snapshot
+    updated_snapshot_df = snapshot_df.copy()
 
-    construction_logger.info(
-        f"{accepted_amendments_df.shape[0]} records amended during construction"
-    )
+    # Validate construction file and drop columns without constructed values
+    validate_data_with_schema(construction_df,
+                              "./config/construction_schema.toml")
+    construction_df = construction_df.dropna(axis="columns", how="all")
 
-    return amended_df
+    # Add flags to indicate whether a row was constructed
+    updated_snapshot_df["is_constructed"] = False
+    construction_df["is_constructed"] = True
 
+    # ! TEMPORARY Fix the year since the snapshot doesn't seem to have one?
+    updated_snapshot_df["year"] = 2022
```

Comment:
There's a function to do this here: outputs.outputs_helpers.create_period_year()

If your code will work with "period_year" as a valid col name, could you create this column here and remove the func from outputs?